Component/input field

iZyadNasser opened the pull request:
![ar-dark-active](https://github.com/user-attachments/assets/45954186-1312-451a-8add-d153186edcd1)
![ar-dark-inactive](https://github.com/user-attachments/assets/c2bea96c-0583-4ecb-9de5-08beedc01f49)
![ar-light-active](https://github.com/user-attachments/assets/d784902e-3200-45b6-a94b-14ad16eb6797)
![ar-light-inactive](https://github.com/user-attachments/assets/5d66ad05-f21b-46cf-9df7-17733472d5fd)
![en-dark-active](https://github.com/user-attachments/assets/0b13dab5-edc0-4859-8d18-56503b2d90da)
![en-dark-inactive](https://github.com/user-attachments/assets/3f744ff4-d406-4895-abf0-d874fe53932e)
![en-light-active](https://github.com/user-attachments/assets/861c8f53-0b4d-42b5-92f1-d7b993e512b0)
![en-light-inactive](https://github.com/user-attachments/assets/1eb57fb6-4590-4515-a64a-a683552d8786)

iZyadNasser committed:
Merge remote-tracking branch 'origin/develop' into common-composables

iZyadNasser committed:
feature: start design of input field and add compose foundation to app so we can run composables there

iZyadNasser committed:
feature: finish input field design

YoussefmSaber commented on design_system/src/main/java/com/cairosquad/design_system/component/InputField.kt:
```diff
@@ -0,0 +1,170 @@
+    modifier: Modifier = Modifier,
+    placeholder: String = "",
+    error: String = "",
+    enabled: Boolean = true,
```
the `error` and `enabled` params you didn't use them inside the composable, why?

iZyadNasser replied:
I have know idea why I forgot about them lol, sorry fixing it now

YoussefmSaber commented on design_system/src/main/java/com/cairosquad/design_system/component/InputField.kt:
```diff
@@ -0,0 +1,170 @@
+                        width = 1.dp,
+                        brush = Brush.horizontalGradient(
+                            listOf(
+                                Color(0xFFEBE6FE),
```
why are you using fixed colors, the colors should be changing wether it's light or dark mode, the colors that were used are the primary and onPrimary colors, please change those,
additionaly instead of creating the brush, make it in part of the design system so you can easily change it.

iZyadNasser replied:
I thought that if the colors are hardcoded in figma it means they are not in design system and are a one time use, I did not know they are the primary and onPrimary, thank you I'll fix it right away

YoussefmSaber replied:
In figma it was hard coded, but this doesn't mean they are, what I did is that i took the color code and matched it with colors we have in our design system and found out that they are primary and onPrimary colors

YoussefmSaber commented on design_system/src/main/java/com/cairosquad/design_system/component/InputField.kt:
```diff
@@ -0,0 +1,170 @@
+}
+
+@Composable
+fun TextFieldIcon(
```
if this composable will not be used in any other composable function, you should make it private.

iZyadNasser replied:
ok

YoussefmSaber reviewed (changes requested):
the gradient should also rotate when the device language become RTL, so the bright side always be next to the leading icon

gehadepada commented on app/src/main/java/com/cairosquad/movio/MainActivity.kt:
```diff
@@ -5,6 +5,7 @@ import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
 
+
```
delete extra space

iZyadNasser replied:
ok

gehadepada commented on design_system/src/main/java/com/cairosquad/design_system/component/InputField.kt:
```diff
@@ -0,0 +1,170 @@
+    modifier: Modifier = Modifier,
+    placeholder: String = "",
+    error: String = "",
+    enabled: Boolean = true,
```
it 's better to make the name of boolean variable as a condition 
ex:isEnabled
 and give more meaningful names for variables like isSingleLineInput, isPasswordField

iZyadNasser replied:
you are right

gehadepada commented on design_system/src/main/java/com/cairosquad/design_system/component/InputField.kt:
```diff
@@ -0,0 +1,170 @@
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+    keyboardActions: KeyboardActions = KeyboardActions.Default,
+) {
+    var isFocused by remember { mutableStateOf(false) }
```
give better name for this variable , i can't understand what this do

iZyadNasser replied:
ok on it

gehadepada commented on design_system/src/main/java/com/cairosquad/design_system/component/InputField.kt:
```diff
@@ -0,0 +1,170 @@
+            .background(Theme.color.surfaces.surfaceContainer)
+            .onFocusChanged { focusState ->
+                isFocused = focusState.isFocused
+            }
```
modifier here should be first parameter

iZyadNasser replied:
are you sure? shouldn't the more important (and required) parameters be first (value and onValueChange are the most important parameters for a text field and both are required

gehadepada commented on design_system/src/main/java/com/cairosquad/design_system/component/InputField.kt:
```diff
@@ -0,0 +1,170 @@
+            }
+            .then(
+                if (isFocused) {
+                    Modifier.border(
```
why you use new Modifier here ? you should use modifier that passed to function

iZyadNasser replied:
No, this is here just to make a conditional modifier, I have already passed the external one above

gehadepada commented on design_system/src/main/java/com/cairosquad/design_system/component/InputField.kt:
```diff
@@ -0,0 +1,170 @@
+                        shape = RoundedCornerShape(8.dp)
+                    )
+                } else {
+                    Modifier
```
same

iZyadNasser replied:
same answer, this is just a place holder for me to write a conditional modifier, it's not write to use the external one in this situation

gehadepada commented on design_system/src/main/java/com/cairosquad/design_system/component/InputField.kt:
```diff
@@ -0,0 +1,170 @@
+                    leadingIcon,
+                    isFocused,
+                    modifier = Modifier.padding(end = 8.dp)
+                )
```
on calling of all composable function i think you should make modifier as first parameter

iZyadNasser replied:
I want to understand your reasoning behind making the modifier first parameter, I think it's better for readability and writability to put the required parameters first then the modifier, but I want to hear you opinion

YoussefmSaber replied:
It's prefered when calling a composable to know what kind of modifiers it's passed on it, but when creating a composable function the modifier should be the First optional parameter, cause in this case the required params should be the first one

iZyadNasser replied:
yeah I know the second rule that it should be the first optional when creating the function, I also usually make it the first optional parameter when I pass it, not "as a rule" but I just think that for readability it's better for any developer reading my code without looking at the function definition to know first the required parameters that the function is not going to work without.

iZyadNasser committed:
feature: fix variable names

iZyadNasser committed:
fix:
- use design system instead of hardcoded colors
- add error handling to text field
- remove redundant qualifier names

iZyadNasser committed:
fix:
- use design system instead of hardcoded colors
- add error handling to text field
- remove redundant qualifier names

YoussefmSaber reviewed (changes requested):
The icon should have a circle ripple effect not as square one, user clip(CircleShape),
![image](https://github.com/user-attachments/assets/555db81c-ac26-4748-8791-b9b08c904278)

and can you provide a sample of arabic version, and make sure that the gradient be affected with the layout direction

iZyadNasser committed:
fix:
- ripple effect of trailing icon

iZyadNasser commented:
Input Field after fixing the gradient direction bug

![ar-fixed](https://github.com/user-attachments/assets/6c0f46ae-b4e9-4918-8948-2da93f61769b)
![en-fixed](https://github.com/user-attachments/assets/6e72c820-aafb-43d4-a984-f80bc376cb7c)

gehadepada commented on design_system/src/main/java/com/cairosquad/design_system/component/InputField.kt:
```diff
@@ -0,0 +1,219 @@
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+    keyboardActions: KeyboardActions = KeyboardActions.Default,
+) {
+    var hasFocus by remember { mutableStateOf(false) }
```
give better name for this variable also

iZyadNasser replied:
what should I call it?
I feel that it is clear enough right now. It means that the input field has focus and can be written on, what are your suggestions?

gehadepada replied:
i don't understand what this do until you explain , so you can name it -> isTextFieldFocused

iZyadNasser committed:
Merge branch 'common-composables' into component/input_field
